src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,55 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
-    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+    private static final DateTimeFormatter DATE_FORMATTER = DateTimeFormatter
+            .ofPattern("dd.MM.yyyy");
+
+    public static String getSalaryInfo(String[] names, String[] data,
+                                       String dateFrom, String dateTo) {
+        LocalDate fromDate = null;
+        LocalDate toDate = null;
+        try {
+            fromDate = LocalDate.parse(dateFrom, DATE_FORMATTER);
+            toDate = LocalDate.parse(dateTo, DATE_FORMATTER);
+        } catch (Exception e) {
+            throw new RuntimeException("Invalid date format");
+        }
+        int[] employeeSalaries = new int[names.length];
+        for (String entry : data) {
+            String[] parts = entry.split(" ");
+            String entryDateStr = parts[0];
+            String employeeName = parts[1];
+            int hoursWorked = Integer.parseInt(parts[2]);
+            int incomePerHour = Integer.parseInt(parts[3]);
```

Comment:
Please instead of hardcoded indexes use constants with proper naming (NAME_INDEX, SALARY_INDEX etc)